riscv/sim.h

```diff
@@ -49,7 +49,7 @@ class sim_t : public htif_t, public simif_t
   void set_remote_bitbang(remote_bitbang_t* remote_bitbang) {
     this->remote_bitbang = remote_bitbang;
   }
-  const char* get_dts() { if (dts.empty()) reset(); return dts.c_str(); }
+  const char* get_dts() { return dts.c_str(); }
```

Comment:
Do you understand why the `reset()` was there before? I don't get it.

Also, it seems nobody calls `reset()` anymore, so that method can be removed entirely.

Reply:
I think its because spike-main calls `sim_t::get_dts` before it runs the htif main thread, which is responsible for calling `sim_t::reset`. 

Before https://github.com/riscv-software-src/riscv-isa-sim/commit/3b26740205c401498fe77c2c9a16c8907a139cd8, the reset here was necessary, as the DTS wouldn't be generated until `make_dtb` in `reset` was called. 

The sequence before https://github.com/riscv-software-src/riscv-isa-sim/commit/3b26740205c401498fe77c2c9a16c8907a139cd8:
 1. `sim_t::sim_t` - does not construct dts/dtb
 2. if `sim_t::get_dts` called, run 4
 3. `sim_t::run` - calls `htif::run`, which calls `sim_t::reset`
 4. `sim_t::reset`  - calls `sim_t::make_dtb` - constructs dts/dtb, queries htif for the reset vec, constructs the rom with reset vec + dtb


The sequence after https://github.com/riscv-software-src/riscv-isa-sim/commit/3b26740205c401498fe77c2c9a16c8907a139cd8:
 1. `sim_t::sim_t` - calls `sim_t::make_dtb` to construct dts + dtb
 2. if `sim_t::get_dts` called, run 4
 3. `sim_t::run` - calls `htif::run`, which calls `sim_t::reset`
 4. `sim_t::reset`  - calls `sim_t::set_rom` - queries htif for the reset vec, constructs the rom with the reset vec embedded

The sequence after this PR becomes
 1. `sim_t::sim_t` - if dtb_enabled, calls `sim_t::make_dtb` to construct dts + dtb,
 2. if `sim_t::get_dts` called, return the dts
 3. `sim_t::run` - calls `htif::run`, which calls `sim_t::reset`
 4. `sim_t::reset` - calls `sim_t::set_rom` - queries htif for the reset vec, constructs the rom with the reset vec embedded
 
 The awkwardness comes from constructing sim_t before htif has determined what the reset vector should be.